dataflow/src/org/checkerframework/dataflow/analysis/FlowExpressions.java

```diff
@@ -179,6 +194,209 @@ public static Receiver internalReprOf(
+                }
+                break;
+            case MEMBER_SELECT:
+                receiver = internalRepOfMemberSelect(provider, (MemberSelectTree) receiverTree);
+                break;
+            case IDENTIFIER:
+                IdentifierTree identifierTree = (IdentifierTree) receiverTree;
+                TypeMirror typeOfId = InternalUtils.typeOf(identifierTree);
+                if (identifierTree.getName().contentEquals("this")
+                        || identifierTree.getName().contentEquals("super")) {
+                    receiver = new ThisReference(typeOfId);
+                    break;
+                }
+                Element ele = TreeUtils.elementFromUse(identifierTree);
+                switch (ele.getKind()) {
+                    case LOCAL_VARIABLE:
+                    case RESOURCE_VARIABLE:
+                    case EXCEPTION_PARAMETER:
+                    case PARAMETER:
+                        receiver = new LocalVariable(ele);
+                        break;
+                    case FIELD:
+                        // Implicit access expression, such as "this" or a class name
+                        Receiver fieldAccessExpression;
+                        TypeMirror enclosingType = ElementUtils.enclosingClass(ele).asType();
+                        if (ElementUtils.isStatic(ele)) {
+                            fieldAccessExpression = new ClassName(enclosingType);
+                        } else {
+                            fieldAccessExpression = new ThisReference(enclosingType);
+                        }
+                        receiver =
+                                new FieldAccess(
+                                        fieldAccessExpression, typeOfId, (VariableElement) ele);
+                        break;
+                }
+        }
+
+        if (receiver == null) {
+            receiver = new Unknown(InternalUtils.typeOf(receiverTree));
+        }
+        return receiver;
+    }
+
+    /**
+     * Returns the implicit receiver of ele.
+     *
+     * <p>Returns either a new ClassName or a new ThisReference depending on whether ele is static
+     * or not. The passed element must be a filed, method, or class.
+     *
+     * @param ele field, method, or class
+     * @return either a new ClassName or a new ThisReference depending on whether ele is static or
+     *     not
+     */
+    public static Receiver internalRepOfImplicitReceiver(Element ele) {
+        TypeMirror enclosingType = ElementUtils.enclosingClass(ele).asType();
+        if (ElementUtils.isStatic(ele)) {
+            return new ClassName(enclosingType);
+        } else {
+            return new ThisReference(enclosingType);
+        }
+    }
+
+    /**
+     * Returns either a new ClassName or ThisReference Receiver for the given tree.
+     *
+     * <p>The Tree should be an expression or a statement that does not have a receiver (or an
+     * implicit receiver). For example, a local variable declaration.
```

Comment:
It seems confusing to say that local variables don't have a receiver, but the method returns a receiver.
It would be nicer if the documentation explained what the pseudo-receiver is and why it makes sense to return one for local variables.